### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by senkiv-oleh**
- [DEMO LINK](https://senkiv-oleh.github.io/layout_html-form/)
- [TEST REPORT LINK](https://senkiv-oleh.github.io/layout_html-form/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,15 +1,144 @@
...
+    <meta
+      name="viewport"
+      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
+    >
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>HTML Form</title>
+    <link rel="stylesheet" href="./style.css">
+  </head>
+
+  <body>
```
Comment (aliceblunt8) [inline src/index.html:15]:
```suggestion
  </head>
  <body>
```

File: src/index.html [hunk 2]
```diff
@@ -1,15 +1,144 @@
...
+    <form
+    action="https://mate-academy-form-lesson.herokuapp.com/create-application"
+     method="post"
+     autocomplete="off"
+     >
```
Comment (aliceblunt8) [inline src/index.html:20]:
```suggestion
    <form
     action="https://mate-academy-form-lesson.herokuapp.com/create-application"
     method="post"
     autocomplete="off"
     >
```

File: src/index.html [hunk 3]
```diff
@@ -1,15 +1,144 @@
...
+      <fieldset>
+        <legend>Personal information</legend>
+
+        <label class="form-field" for="surnmae">
+          Surname:
+          <input type="text" id="surnmae"
+            name="surnmae" required>
+        </label>
```
Comment (aliceblunt8) [inline src/index.html:28]: 3 or more than 3 attributes -> 1 attribute per 1 line
fix everywhere

```suggestion
        <label class="form-field" for="surnmae">
          Surname:
          <input
            type="text"
            id="surname"
            name="surname"
            required
          >
        </label>
```

File: src/index.html [hunk 4]
```diff
@@ -1,15 +1,144 @@
...
+          Do you love cats?
+          <label for="yes">
+            <input type="radio" name="cat"
+              id="yes">Yes
+          </label>
+
+          <label for="no">
+            <input type="radio" name="cat"
+              id="no">No
+          </label>
```
Comment (aliceblunt8) [inline src/index.html:89]: the name for radio buttons must be clear
this data will be sent to the server in such format: name(cat): value (yes or no), is it clear for you what does it mean?
better give a num like this -> love-cats: yes or no

File: src/index.html [hunk 5]
```diff
@@ -1,15 +1,144 @@
...
+        </div>
+
+        <label class="form-field" for="f-color">
+          What is your favorite color?
+          <input type="color" name="f-color"
+            id="f-color">
```
Comment (aliceblunt8) [inline src/index.html:95]:
```suggestion
          <input type="color" name="favorite-color"
            id="color">
```

File: src/index.html [hunk 6]
```diff
@@ -1,15 +1,144 @@
...
+        </label>
+
+        <label class="form-field" for="appt-time">
+          What time do you go to bed?
+          <input  type="time" step="2"
+            name="appt-time" id="appt-time">
+        </label>
```
Comment (aliceblunt8) [inline src/index.html:102]: appt-time? what does it mean?

File: src/index.html [hunk 7]
```diff
@@ -1,15 +1,144 @@
...
+            <option value="lada">Lada</option>
+          </select>
+        </label>
+
+        <label class="form-field" for="cowbell">
+          How do you rate our work?
+          <input type="range" name="cowbell"
+            min="0" max="100"
+            id="cowbell">
+        </label>
```
Comment (aliceblunt8) [inline src/index.html:119]: cowbell?!

File: src/index.html [hunk 8]
```diff
@@ -1,15 +1,144 @@
...
+        <label class="form-field" for="comments">
+          Comments:
+          <textarea name="comments" id="comments"></textarea>
+        </label>
+
+        <label class="form-field" for="recommend">
+          Would you recommend us?
+          <select name="recommend" id="recommend">
+            <option value="Yes" selected>Yes</option>
+            <option value="No">No</option>
```
Comment (aliceblunt8) [inline src/index.html:134]:
```suggestion
            <option value="yes" selected>Yes</option>
            <option value="no">No</option>
```

File: src/index.html [hunk 9]
```diff
@@ -1,15 +1,144 @@
...
+          </select>
+        </label>
+      </fieldset>
+
+      <button type="submit" onsubmit="onSubmit()">Submit</button>
```
Comment (aliceblunt8) [inline src/index.html:139]: type="submit" is a default type for buttons
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Comment by senkiv-oleh** [comment]
ok

ср, 24 бер. 2021 о 10:31 Alisa Ivashkevych ***@***.***> пише:

File: src/index.html [hunk 10]
```diff
@@ -1,15 +1,203 @@
...
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>HTML Form</title>
+    <link rel="stylesheet" href="./style.css">
+  </head>
+  <body>
+    <form
+    action="https://mate-academy-form-lesson.herokuapp.com/create-application"
+    method="post"
+    autocomplete="off"
+    >
```
Comment (aliceblunt8) [inline src/index.html:19]: 2 spaces

```suggestion
    <form
      action="https://mate-academy-form-lesson.herokuapp.com/create-application"
      method="post"
      autocomplete="off"
    >
```

File: src/index.html [hunk 11]
```diff
@@ -1,15 +1,203 @@
...
+      <fieldset>
+        <legend>Personal information</legend>
+
+        <label class="form-field" for="surnmae">
+          Surname:
+          <input
+            type="text"
+            id="surnmae"
+            name="surnmae"
```
Comment (aliceblunt8) [inline src/index.html:28]:
```suggestion
            id="surname"
            name="surname"
```

File: src/index.html [hunk 12]
```diff
@@ -1,15 +1,203 @@
...
+          </label>
+
+          <label for="hate-cats">
+            <input
+              type="radio"
+              id="hate-cats"
+              name="no"
+              value="no"
+            >No
+          </label>
```
Comment (aliceblunt8) [inline src/index.html:125]: name for a group of radio buttons should be the same, rename to 'love-cats', cause now it's possible to choose 2 options at a time
```suggestion
        <div class="form-field" for="cats">
          Do you love cats?
          <label for="love-cats">
            <input
              type="radio"
              id="love-cats"
              name="love-cats"
              value="yes"
            >Yes
          </label>

          <label for="hate-cats">
            <input
              type="radio"
              id="hate-cats"
              name="love-cats"
              value="no"
            >No
          </label>
```

File: src/index.html [hunk 13]
```diff
@@ -1,15 +1,203 @@
...
+
+        <label class="form-field" for="rate-work">
+          How do you rate our work?
+          <input
+            type="range"
+            name="rate-work"
+            value="0"
+            id="rate-work"
+          >
+        </label>
```
Comment (aliceblunt8) [inline src/index.html:168]: min? max?

File: src/index.html [hunk 14]
```diff
@@ -1,15 +1,203 @@
...
+      </fieldset>
+
+      <button
+        type="submit"
+        value="Submit"
+        onsubmit="onSubmit()"
+      >
+        Submit
+      </button>
```
Comment (aliceblunt8) [inline src/index.html:197]: type="submit" is a default type for buttons
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

```suggestion
      <button onsubmit="onSubmit()">
        Submit
      </button>
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 14 above)
Comment (senkiv-oleh) [reply src/index.html]: When I did this, I saw the test not running.

reason:  Element not found for capturing: [type="submit"]

**Pull request closed without merging**